99robots-header-footer-code-manager.php

```diff
@@ -1001,6 +1016,421 @@ public static function hfcm_request_handler()
+                            $selectizeResults[] = array('value' => $pdata->ID, 'text' => sanitize_text_field( $pdata->post_title ), 'disabled' => true);
+                        }
+                        else {
+                            $selectOptions .= '<option class="left-side-option clone-button button-id-'.$pdata->ID.'" value="'.$pdata->ID.'|'.$pdata->post_title.'">'.sanitize_text_field( $pdata->post_title ).' + </option>';
+                            $selectizeResults[] = array('value' => $pdata->ID, 'text' => sanitize_text_field( $pdata->post_title ));
+                        }
+        
+                    }
+                    else {
+                        if( is_array( $_POST['disabledOptions'] ) && !empty( $_POST['disabledOptions'] ) && in_array( $pdata->ID, $_POST['disabledOptions'] ) ){
+                            $selectOptions .= '<option class="left-side-option clone-button button-id-'.$pdata->ID.'" value="'.$pdata->ID.'|'.$pdata->post_title.'" disabled>'.sanitize_text_field( $pdata->post_title ).' + </option>';
+                            $selectizeResults[] = array('value' => $pdata->ID, 'text' => sanitize_text_field( $pdata->post_title ), 'disabled' => true);
+                        }
+                        else {
+                            $selectOptions .= '<option class="left-side-option clone-button button-id-'.$pdata->ID.'" value="'.$pdata->ID.'|'.$pdata->post_title.'">'.sanitize_text_field( $pdata->post_title ).' + </option>';
+                            $selectizeResults[] = array('value' => $pdata->ID, 'text' => sanitize_text_field( $pdata->post_title ));
+                        }
+                    }
+                    
+        
+                }
+                $json_output['posts'] = $selectOptions;
+                $json_output['count'] = count($posts);
+                $json_output['selectize_posts'] = $selectizeResults;
+                echo wp_json_encode( $json_output );
+                wp_die();
+            }
+        }
+
+        public static function hfcm_request_handler_custom_post_type() {
+            
+            // check user capabilities.
+            $nnr_hfcm_can_edit = current_user_can( 'manage_options' );
+
+            if ( !$nnr_hfcm_can_edit ) {
+                echo 'Sorry, you do not have access to this page.';
+                return false;
+            }
+
+        
+            if ( isset( $_POST['insert'] ) ) {
+                // Check nonce
+                check_admin_referer( 'create-snippet' );
+            } else {
+                if ( ! isset( $_REQUEST['id'] ) ) {
+                    die( 'Missing ID parameter.' );
+                }
+                $id = (int) $_REQUEST['id'];
+            }
+
+            if ( isset( $_POST['update'] ) ) {
+                // Check nonce
+                check_admin_referer( 'update-snippet_' . $id );
```

Comment:
https://developer.wordpress.org/reference/functions/check_admin_referer/
why aren't we doing anything with this?

Reply:
fix these basic issues throughout